### PR TITLE
Adds mi-go encampment terrain and overmap

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds new basic alien terrain types and a basic mi-go encampment map, does not yet spawn in-game."```

#### Purpose of change
The mi-go are meant to be an entire faction in-game, and there is a significant amount of lore around them, but currently they only exist as isolated minor encounters.

#### Describe the solution
This adds the infrastructure and early map designs for a mi-go encampment. This includes some very basic furniture types, with some uses for them, and nested mapgen so that in the future alternative map styles can be used.

This uses the brand-new ability for furniture to be field emitters, so that several furniture items in the base emit heat and occasional clouds of toxic gas, making it comfortable for aliens but not humans. These should offer an unusual challenge to explorers... I don't think anywhere else in the game forces you to deal with quite this severe of heat stroke while fighting very powerful enemies.

More furnishings for the encampment still need to be added, as well as new mi-go enemy types and loot that can be found.

#### Describe alternatives you've considered
I considered a more hive-like structure but the smooth organic lines wound up looking better.

#### Additional context
Upcoming, over several PRs:
- add mi-go furnishings
- add nested mapgen for primary map
- add mi-go surgeon, mi-go guard, and mi-go slaver enemy types
- add lootable resin and recipes for constructing resin walls and floors with it

A balmy 56 degrees celsius, with clouds of toxic gas wafting out of the gaspers. Nice.
image

Ground floor, full map
image

Exploring in the dark without a clairvoyance artifact has a much better feel
image

The spire, leading to the landing platform.
image

The landing platform.
image

### CODE REVIEW

**Comment by Mecares** [comment]
Will the slaver get nets and bolas from #32094?

**Comment by I-am-Erk** [comment]
Haha no, but they will have disabling attacks

File: data/json/furniture.json [hunk 1]
```diff
@@ -6697,6 +6697,103 @@
...
+    "required_str": -1,
+    "flags": [ "TRANSPARENT", "EMITTER", "NONFLAMMABLE" ],
+    "emissions": [ "emit_glimmer" ],
+    "bash": { "str_min": 8, "str_max": 20, "sound": "splorch!", "sound_fail": "whump!" }
+  },
+  {
+    "type": "furniture",
+    "id": "f_alien_anemone",
+    "name": "wafting anemone",
+    "description": "A fleshy white protruberance growing from the floor, with a cluster of tendrils pouring out of it.  It looks almost exactly like a sea anemone, even waving gently as though in the water.",
```
Comment (Night-Pryanik) [inline data/json/furniture.json]:
```suggestion
    "description": "A fleshy white protuberance growing from the floor, with a cluster of tendrils pouring out of it.  It looks almost exactly like a sea anemone, even waving gently as though in the water.",
```

File: data/json/furniture.json [hunk 2]
```diff
@@ -6697,6 +6697,103 @@
...
+    "deployed_item": "alien_pod_resin",
+    "examine_action": "deployed_furniture",
+    "flags": [ "TRANSPARENT", "NONFLAMMABLE" ],
+    "bash": { "str_min": 25, "str_max": 60, "sound": "splorch!", "sound_fail": "whump." }
+  },
+  {
+    "type": "furniture",
+    "id": "f_alien_table",
+    "name": "fleshy altar",
+    "description": "This pulsing protruberance juts from the floor, its sides covered in scaled, oozing skin. The surface is flat, but undulates softly.  A handful of unidentifiable appendages reach from the sides, suggesting a sort of nightmarish living autodoc.",
```
Comment (Night-Pryanik) [inline data/json/furniture.json]:
```suggestion
    "description": "This pulsing protuberance juts from the floor, its sides covered in scaled, oozing skin. The surface is flat, but undulates softly.  A handful of unidentifiable appendages reach from the sides, suggesting a sort of nightmarish living autodoc.",
```

**Comment by I-am-Erk** [comment]
Jenkins Rebuild

**Pull request merged**